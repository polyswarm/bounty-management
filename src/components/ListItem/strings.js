const strings = {
  remove: 'remove'
};

export default strings;
