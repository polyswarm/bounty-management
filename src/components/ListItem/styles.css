.ListItem {
  background-clip: border-box;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  height: 48px;
  line-height: 46px;
  justify-content: space-between;
  position: relative;
  width: 100%;
}

.ListItem:hover {
  background-color: rgba(0, 0, 0, .1);
}

.active {
  color: rgb(46, 255, 109);
  font-weight: bold;
}

.alternate {
  background-color: rgba(0, 0, 0, .2);
  color: #8500ff;
  font-weight: bold;
}

.alert {
  background-color: rgb(46, 255, 109);
  display: inline-block;
  height: 100%;
  width: 10px;
}

.ListItem-Child {
  box-sizing: border-box;
  display: inline-block;
  height: 100%;
  left: 0;
  margin: 0;
  padding: 0 20px;
  position: absolute;
  right: 0;
  top: 0;
}

.redx {
  background: url('/img/red-x.svg');
  height: 48px;
}
