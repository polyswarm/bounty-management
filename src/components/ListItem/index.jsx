import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import RemoveButton from '../RemoveButton';
// Component imports
import strings from './strings';
import './styles.css';

class ListItem extends Component {
  constructor(props) {
    super(props);
    this.onClickHandler = this.onClickHandler.bind(this);
  }

  render () {
    const { props: { item } } = this;
    return(
      <div className='ListItem'>
        <p className='ListItem-Name'>{item}</p>
        <span className='ListItem-Remove'>
          <RemoveButton onClick={this.onClickHandler}>
            {strings.remove}
          </RemoveButton>
        </span>
      </div>
    );
  }

  onClickHandler() {
    const { props: { remove } } = this;
    if (remove) {
      remove();
    }
  }
}

ListItem.proptypes = {
  item: PropTypes.string.isRequired,
  remove: PropTypes.func.isRequired,
};

export default ListItem;
