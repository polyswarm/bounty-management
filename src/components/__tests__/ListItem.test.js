import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson, mountToJson} from 'enzyme-to-json';
import ListItem from '../ListItem';

it('renders without crashing', () => {
  const item = 'asdf';
  const wrapper = render(<ListItem item={item} />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('calls the mock function when I click close', () => {
  const remove = jest.fn();
  const item = 'asdf';
  const wrapper = mount(<ListItem item={item} remove={remove} />);
  wrapper.find('.ListItem').simulate('mouseEnter');
  wrapper.find('button').simulate('click');
  expect(remove).toHaveBeenCalled();
});

it('shows remove button when hovered', () => {
  const remove = jest.fn();
  const item = 'asdf';
  const wrapper = mount(<ListItem item={item} remove={remove} />);
  wrapper.find('.ListItem').simulate('mouseEnter');
  expect(wrapper.find('button')).toHaveLength(1);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('adds the active class when active property is true', () => {
  const remove = jest.fn();
  const item = 'asdf';
  const wrapper = render(<ListItem item={item} remove={remove} active/>);
  expect(wrapper.hasClass('active')).toBeTruthy();
  expect(renderToJson(wrapper)).toMatchSnapshot();
});
