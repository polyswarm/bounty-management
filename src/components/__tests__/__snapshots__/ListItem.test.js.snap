// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds neither extra class when just alternate property is true 1`] = `
<li
  class="ListItem"
>
  <div>
    <span
      class="ListItem-Child"
    >
      asdf
    </span>
  </div>
</li>
`;

exports[`adds the active class when active property is true 1`] = `
<li
  class="ListItem active"
>
  <div>
    <span
      class="ListItem-Child"
    >
      asdf
    </span>
  </div>
</li>
`;

exports[`adds the alternate class when active and alternate properties are true 1`] = `
<li
  class="ListItem alternate"
>
  <div>
    <span
      class="ListItem-Child"
    >
      asdf
    </span>
  </div>
</li>
`;

exports[`renders the child I pass 1`] = `
<li
  class="ListItem"
>
  <div>
    <span
      class="ListItem-Child"
    >
      asdf
    </span>
  </div>
</li>
`;

exports[`renders without crashing 1`] = `
<li
  class="ListItem"
>
  <div>
    <span
      class="ListItem-Child"
    >
      asdf
    </span>
  </div>
</li>
`;

exports[`shows remove button when hovered 1`] = `
<ListItem
  remove={[Function]}
>
  <li
    className="ListItem"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div>
      <span
        className="ListItem-Child"
        onClick={[Function]}
      >
        asdf
      </span>
    </div>
    <span
      className="ListItem-Remove"
    >
      <RemoveButton
        onClick={[Function]}
      >
        <button
          className="Remove-Button"
          onClick={[Function]}
        >
          <img
            alt="remove"
            className="redx"
            src="../public/img/red-x.svg"
          />
        </button>
      </RemoveButton>
    </span>
  </li>
</ListItem>
`;

exports[`shows the badge when alert is set 1`] = `
<li
  class="ListItem active"
>
  <div>
    <div
      class="alert"
    >
      
    </div>
    <span
      class="ListItem-Child"
    >
      asdf
    </span>
  </div>
</li>
`;
